### PR TITLE
Overhaul the Preferences

### PR DESCRIPTION
**Pull request by shun-iwasawa**
**PLEASE NOTE: I need this PR to be carefully tested before merging, as I may miss something!**

This refactoring PR will overhaul the preferences.
The source code for the preferences was very messy, having separate similar functions for every variables' I/O.
I tried to clear them up by putting almost all the preference values in one container (Id-QVariant map). Now, I managed to reduce 2,000+ lines of code!

I roughly tested and find that the change has (hopefully) no bad side effect, but I would like this to have further tests from other reviewers before merging.

### CODE REVIEW

**Comment by RodneyBaker** [comment]
Travis seems to have failed building on the last two PRs.  
It's not clear if that is due to the PRs themselves or Travis itself.

**Comment by manongjohn** [comment]
There is an issue with the macos builds that started yesterday. It's not using the brew installed GLEW library and the default stuff has changed somehow.

I noticed it on my local build environment that emulates Travis. Looking into it as it will affect OTX builds.

File: toonz/sources/toonzlib/preferences.cpp [hunk 1]
```diff
@@ -563,675 +319,491 @@ Preferences::Preferences()
...
 }
 
 //-----------------------------------------------------------------
 
-void Preferences::enablePreviewAlwaysOpenNewFlip(bool on) {
-  m_previewAlwaysOpenNewFlipEnabled = on;
-  m_settings->setValue("previewAlwaysOpenNewFlip", on ? "1" : "0");
+PreferencesItem &Preferences::getItem(const PreferencesItemId id) {
+  assert(m_items.contains(id));
+  if (!m_items.contains(id)) return PreferencesItem();
```
Comment (manongjohn) [inline toonz/sources/toonzlib/preferences.cpp]: Linux and macOS builds are failing here with this message

/Users/travis/build/opentoonz/opentoonz/toonz/sources/toonzlib/preferences.cpp:668:37: error: non-const lvalue reference to type 'PreferencesItem' cannot bind to a temporary of type 'PreferencesItem'
  if (!m_items.contains(id)) return PreferencesItem();

Comment (shun-iwasawa) [reply toonz/sources/toonzlib/preferences.cpp]: Thank you for the notice, I've made the same mistake again.
I removed the problematic line.

**Comment by RodneyBaker** [comment]
I've been testing this PR (from user POV via appveyor portable) as I get the chance and haven't noticed anything odd, out of place or non-functioning.  

Off topic:  There are a number of things that need to be addressed with Preferences that would fit more into a second 'Reorganize the Prefererences' PR.  Most of these fit into general refinement category although there are a few cases of things meant to work that do not; example:  the cursor changes when hovering over the popup window borders and corners to suggest Preferences window can be resized but window cannot be resized/rescaled.

**Comment by gab3d** [comment]
@RodneyBaker 
...but the Preferences window can be resized, at least on Windows. 🤔

**Comment by manongjohn** [comment]
I can resize Preferences. 

I did notice sometimes the cursor on mouse overs doesn't change to show a pane's edge can be moved until after you try to move it. I think it happened with Style Editor and Level Editor, the 2 frames that are designed to keep size when main window resizes. I haven't seen that with other windows though.

**Comment by RodneyBaker** [comment]
Interesting.
Apparently I was on the Animation tab and attempting to make the window smaller and it is/was constrained to be as large as the contents of other tabs.  

Attempt to make the Preferences window as small as possible while the Animation tab is open and see how small you can make the window.  At a guess I'd say there is a minimum size allowable to accommodate the other tabs.

**Comment by shun-iwasawa** [comment]
>Attempt to make the Preferences window as small as possible while the Animation tab is open and see how small you can make the window. At a guess I'd say there is a minimum size allowable to accommodate the other tabs.

Yes, it's an expected behavior of `QStackedWidget` used for the stacked pages in the popup... You need to nest it in `QScrollArea` to make it shrinkable smaller than the minimum size.
It's similar to the issue #2794 regarding the output settings popup.

I further tested this PR in our studio and confirmed that it works OK with 3DLUT settings, which is quite special among others. 
Thank you for testing, I'll merge this.

**Comment by manongjohn** [comment]
@shun-iwasawa Somehow this PR is not keeping History correctly. It only keeps the last action. Can only undo the last action.

**Comment by shun-iwasawa** [comment]
@manongjohn I'll check it right away!

File: toonz/sources/toonzlib/preferences.cpp [hunk 2]
```diff
@@ -219,287 +208,28 @@ bool Preferences::LevelFormat::matches(const TFilePath &fp) const {
...
-  getValue(*m_settings, "viewShrink", m_shrink);
-  getValue(*m_settings, "viewStep", m_step);
-  getValue(*m_settings, "blanksCount", m_blanksCount);
-  getValue(*m_settings, "askForOverrideRender", m_askForOverrideRender);
-  r = 255, g = 255, b = 255;
-  getValue(*m_settings, "blankColor.r", r);
-  getValue(*m_settings, "blankColor.g", g);
-  getValue(*m_settings, "blankColor.b", b);
-  getValue(*m_settings, "undoMemorySize", m_undoMemorySize);
-  setUndoMemorySize(m_undoMemorySize);
```
Comment (shun-iwasawa) [inline toonz/sources/toonzlib/preferences.cpp]: Ah! I missed this line in ctor!

**Comment by shun-iwasawa** [comment]
@manongjohn Fixed the problem. Thank you so much for the notice!

**Pull request merged**